### PR TITLE
Sign-in view

### PR DESCRIPTION
**Pull request by adshayanB**
image

### CODE REVIEW

**Review by RishanR — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                        loggedIn.value = "True"
+
+//                    val intent = Intent(this, MainActivity::class.java)
+//                    startActivity(intent)
+                } else {
+                    loggedIn.value = "False"
+                        Log.v("GG", "Did not wor")
+
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (RishanR) [reply app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt:74]: Sounds good

File: app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class SignInViewModel @Inject constructor(
+    userRepository: UserRepository
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt:31]: Change this to `private val userRepository: UserRepository`

File: app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    ))
+    val state: StateFlow<SignInModel.SignInViewState>
+        get() = _state
+
+
+    private val username: MutableStateFlow<String> = MutableStateFlow(_state.value.userName)
+    private val password: MutableStateFlow<String> = MutableStateFlow(_state.value.passWord)
+    private val loggedIn: MutableStateFlow<String> = MutableStateFlow(_state.value.loggedIn)
+    private val isLoading: MutableStateFlow<Boolean> = MutableStateFlow(_state.value.isLoading)
+    private val userRepository: UserRepository = userRepository;
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_in/SignInViewModel.kt:44]: You can get rid of this line with the above change

File: app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+//    ) {
+//        Text(state.selectedTab.toString())
+//    }
+    Box(modifier = Modifier.fillMaxSize()) {
+        ClickableText(
+            text = AnnotatedString("Sign up here"),
+            modifier = Modifier
+                .align(Alignment.BottomCenter)
+                .padding(20.dp),
+            onClick = { },
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt:65]: Add a TODO comment here so we know that we need to add the onClick for this later

File: app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        TextField(
+            label = { Text(text = "Password") },
+            value = password.value,
+            visualTransformation = PasswordVisualTransformation(),
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+            onValueChange = { password.value = it })
+
+        Spacer(modifier = Modifier.height(20.dp))
+        Box(modifier = Modifier.padding(40.dp, 0.dp, 40.dp, 0.dp)) {
+            Button(
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt:100]: Use the ButtonView component in uiComponents

**Review by RishanR — CHANGES_REQUESTED** [review]
Just a few more requested changes above, but its looking good!

**Comment by adshayanB** [comment]
Will use a similar format to implement the sign up screen once this pr is merged :)

**Review by RishanR — APPROVED** [review]
LGTM!

**Pull request merged**